lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystemInputFile.java

```diff
@@ -0,0 +1,170 @@
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.runtime.NotFoundRuntimeException;
+import alluxio.grpc.OpenFilePOptions;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoInput;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Optional;
+
+import static io.trino.filesystem.alluxio.AlluxioUtils.convertToAlluxioURI;
+import static java.util.Objects.requireNonNull;
+
+public class AlluxioFileSystemInputFile
+        implements TrinoInputFile
+{
+    private final Location location;
+
+    private final FileSystem fileSystem;
+
+    private final String mountRoot;
+    private Optional<Instant> lastModified;
+
+    private Long length;
+
+    private URIStatus status;
+
+    public AlluxioFileSystemInputFile(Location location, Long length, FileSystem fileSystem, String mountRoot, Optional<Instant> lastModified)
+    {
+        this.location = requireNonNull(location, "location is null");
+        this.fileSystem = requireNonNull(fileSystem, "fileSystem is null");
+        this.mountRoot = requireNonNull(mountRoot, "mountRoot is null");
+        this.length = length;
+        this.lastModified = requireNonNull(lastModified, "lastModified is null");
+    }
+
+    @Override
+    public TrinoInput newInput()
+            throws IOException
+    {
+        try {
+            return new AlluxioFileSystemInput(openFile(), this);
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error newInput() file: %s".formatted(location), e);
+        }
+    }
+
+    @Override
+    public TrinoInputStream newStream()
+            throws IOException
+    {
+        try {
+            return new AlluxioTrinoInputStream(location, openFile(), getStatus());
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error newStream() file: %s".formatted(location), e);
+        }
+    }
+
+    private FileInStream openFile()
+            throws IOException, AlluxioException
+    {
+        if (!exists()) {
+            throw new FileNotFoundException("File does not exist: " + location);
+        }
+        return fileSystem.openFile(getStatus(), OpenFilePOptions.getDefaultInstance());
+    }
+
+    private URIStatus getStatus(boolean lazy)
+            throws IOException
+    {
+        if (lazy) {
+            if (status == null) {
+                getStatus();
+            }
+            return status;
+        }
+        return getStatus();
+    }
+
+    private URIStatus getStatus()
+            throws IOException
+    {
+        try {
+            //TODO: create a URIStatus object based on the location field
+            status = fileSystem.getStatus(convertToAlluxioURI(location, mountRoot));
+        }
+        catch (FileDoesNotExistException | NotFoundRuntimeException e) {
+            return null;
```

Comment:
`throw new FileNotFoundException("File does not exist: %s".formatted(location));` here instead of handling null elsewhere

Reply:
resolved